news/13943-envvars-force-uppercase

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Add an `envvars_force_uppercase` setting defaulting to `True`. (#13713 via #13943)
```

Comment:
This line could use a bit more detail so that folks can understand what exactly got added when reading through the release changelog. Maybe something like:
```suggestion
* Add an `envvars_force_uppercase` setting which defaults to `True`, uppercasing all environment variables; when `envvars_force_uppercase` is set to `False`, conda will only save preserved-case variable names. (#13713 via #13943)
```